### PR TITLE
TDI-37658: avoid the NPE when use tsalesforcebulkexec and

### PR DESCRIPTION
**Pull request by wwang-talend**
https://jira.talendforge.org/browse/TDI-37658

### CODE REVIEW

**Comment by francisu** [comment]
unit test please.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/298/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/298/)

File: components-salesforce/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecDefinition.java [hunk 1]
```diff
@@ -62,11 +62,7 @@ public boolean isConditionalInputs() {
 
     @Override
     public RuntimeInfo getRuntimeInfo(Properties properties, ConnectorTopology componentType) {
-        if (componentType == ConnectorTopology.OUTGOING) {
-            return getCommonRuntimeInfo(this.getClass().getClassLoader(), SalesforceSource.class);
-        } else {
-            return null;
-        }
+        return getCommonRuntimeInfo(this.getClass().getClassLoader(), SalesforceSource.class);
```
Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecDefinition.java]: yes

Comment (gonchar-ivan) [inline components-salesforce/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecDefinition.java]: As for caller should handle null value:
RuntimeInfo runtimeInfo = definition.getRuntimeInfo(properties, componentType);
if (runtimeInfo == null) {
   throw new Exception("Please, use correct connection"); 
}

So, for my opinion there is a bug in Studio side. What do you think about it?

Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecDefinition.java]: agree, but this is some risk for that as the getRuntimeInfo method will be called by many places, not only the job runtime. This is a quick fix, we can do it later.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/302/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/302/)

File: components-salesforce/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecDefinition.java (hunk 1 above)
Comment (gonchar-ivan) [inline components-salesforce/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecDefinition.java]: Even javadoc says that such method may return null
https://github.com/Talend/components/blob/master/components-api/src/main/java/org/talend/components/api/component/ComponentDefinition.java#L117-L128

;-)

**Comment by wwang-talend** [comment]
Hi Ivan, how about the last commit? it follow the document.

**Comment by gonchar-ivan** [comment]
ok for me

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/305/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/305/)

**Pull request merged**